h2/events.py

```diff
@@ -136,6 +136,52 @@ def __repr__(self):
         )
 
 
+class _HeadersSent(object):
+    """
+    The _HeadersSent event is fired whenever headers are received.  This event
+    carries the HTTP headers and the stream ID of the new stream.
+
+    This is an internal event, used to determine validation steps on
+    outgoing header blocks.
+    """
+    def __init__(self):
+        #: The Stream ID for the stream this response was made on.
+        self.stream_id = None
+
+        #: The headers.
+        self.headers = None
```

Comment:
This ordinarily gets populated by the stream object itself, because the state machine doesn't get access to the headers. We don't need it here. =) Nor do we even really need the stream ID.